### PR TITLE
First pass at django based API

### PR DESCRIPTION
**Pull request by kaitlin**
Includes the basic scaffolding for a django rest framework-based API. 

Currently it's just one endpoint with a scant model and a bit of mocked data is at
http://localhost:8000/api/v1/awards/

@cmc333333 would be grateful for you to take a look, it's been a while since I had to set up a Django project so my conventions may be a bit out of date :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,118 @@
...
+1. Download the correct PostgreSQL installer for your operating system from [EnterpriseDB](http://www.enterprisedb.com/products-services-training/pgdownload) (we recommend PostgreSQL 9.4.x).
+2. Run the installer. As you proceed through the installation wizard, note your choices for port number, username, and password. You will need those when creating the broker's configuration file.
+
+More complete install documentation is available on the PostgreSQL [wiki](https://wiki.postgresql.org/wiki/Detailed_installation_guides).
+
+**Note:** If you're setting up the broker on Mac OSX, we recommend using [homebrew](http://brew.sh) to install PostgreSQL.
+
+### Install Python and Create Virtual Environment
+
+The broker's backend components currently run on Python 3.x. These instructions will walk you through the process of installing Python and creating a Python-based virtual environment to house the API. A virtual environment will isolate the broker software and its libraries from those running on your local system and prevent potential conflicts.
```
Comment (cmc333333) [inline README.md]: "The broker" is mentioned a few times in this doc, which I don't _think_ you want?

Comment (kaitlin) [inline README.md]: derp, nope. I cribbed heavily. Will update.

File: README.md [hunk 2]
```diff
@@ -1 +1,118 @@
...
+    Operations to perform:
+    Apply all migrations: admin, auth, awards, contenttypes, sessions
+    Running migrations:
+    Applying awards.0001_initial... OK
+
+Once you've done that, you're ready to start the development server
+
+    $ python manage.py runserver
+
+This will run the application at `0.0.0.0:8000` by default. You can change that by passing the host and port number in to the `runserver` command.
```
Comment (cmc333333) [inline README.md]: Minor, but `127.0.0.1` is slightly different than `0.0.0.0` (the latter can be accessed from outside localhost, which is important for virtual machines). This should be the former.

File: usaspending_api/awards/tests.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from django.test import TestCase, Client
+
+class AwardTests(TestCase):
+
+    #load the test fixture
+    fixtures = ['awards']
```
Comment (cmc333333) [inline usaspending_api/awards/tests.py]: Super minor, but I've never been a fan of loading fixtures from a file -- it requires the test readers look in multiple places to understand what's being tested. 

Out of scope for this PR, but something I'd recommend looking at: the sooner you get `py.test` in, the sooner you'll see it's benefits. It's completely compatible with unittest-style stuff, but I don't think you'll want to go back.

File: usaspending_api/awards/tests.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    def setUp(self):
+        self.c = Client()
```
Comment (cmc333333) [inline usaspending_api/awards/tests.py]: I believe django's `TestCase` comes with a `self.client`

File: usaspending_api/local_settings.example.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': 'usaspending_api',
+        'USER': 'usaspending_api',
+        'PASSWORD': '',
+    }
+}
```
Comment (cmc333333) [inline usaspending_api/local_settings.example.py]: Not a big issue, but something to consider earlier on -- I get the sense that the `local_settings` approach is now frowned on in favor of environment variables (in part due to the [12-factor push](https://12factor.net/)). In the database scenario, we tend to use [`dj-database-url`](https://github.com/kennethreitz/dj-database-url) on cloud.gov projects; it effectively expands a single database string into these four fields.

File: README.md (hunk 2 above)
Comment (kaitlin) [inline README.md]: good catch

File: usaspending_api/awards/tests.py (hunk 3 above)
Comment (kaitlin) [inline usaspending_api/awards/tests.py]: cool. I will work on it this afternoon. Thanks!

File: usaspending_api/awards/tests.py (hunk 4 above)
Comment (kaitlin) [inline usaspending_api/awards/tests.py]: so it does!

File: usaspending_api/local_settings.example.py (hunk 5 above)
Comment (kaitlin) [inline usaspending_api/local_settings.example.py]: yeah I had the vague feeling as I was writing it that it was going to date me. I'll work on updating this for this PR.

File: usaspending_api/settings.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/1.10/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = '&=pi7xiahctpu@=nsp9!9qec6k(jcn##)b@u&s*sx9%kp+m66o'
```
Comment (cmc333333) [inline usaspending_api/settings.py]: This'll throw warnings when/if you integrate [`bandit`](https://github.com/openstack/bandit) in the future. One approach that works relatively well is to generate a new secret key (e.g. via `django.utils.crypto.get_random_string`) whenever the app is restarted. The downside to this approach is that your user will get logged out.

File: usaspending_api/settings.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+
+ROOT_URLCONF = 'usaspending_api.urls'
+
+TEMPLATES = [
+    {
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
```
Comment (cmc333333) [inline usaspending_api/settings.py]: No doubt a style thing, but I _think_ the community's moving towards jinja2 for templates. If you prefer them, now'd be a good time to set the standard.

File: usaspending_api/local_settings.example.py (hunk 5 above)
Comment (kaitlin) [inline usaspending_api/local_settings.example.py]: how do people generally manage this locally? autoenv or virtualenvwrapper post activate hooks?

**Comment by cmc333333** [comment]
@kaitlin all good stuff! I threw in random comments, but I think this is good to merge whenever you're comfortable.

**Comment by cmc333333** [comment]
Oh also, probably out of scope for this PR, but setting up automated tests, pep8 testing, and the like would all be good to do upfront, too

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,7 @@
+Django==1.10.1
+django-debug-toolbar==1.5
+django-filter==0.14.0
+djangorestframework==3.4.6
+Markdown==2.6.6
+psycopg2==2.6.2
+sqlparse==0.2.1
```
Comment (cmc333333) [inline requirements.txt]: :clap: :clap: :clap: for pinning your dependencies

File: usaspending_api/local_settings.example.py (hunk 5 above)
Comment (cmc333333) [inline usaspending_api/local_settings.example.py]: I think autoenv is a good approach, but I'm definitely on poor footing here and don't practice what I'm preaching 😄

File: usaspending_api/settings.py (hunk 6 above)
Comment (kaitlin) [inline usaspending_api/settings.py]: updated

File: usaspending_api/settings.py (hunk 7 above)
Comment (kaitlin) [inline usaspending_api/settings.py]: yeah I think I'll leave for now since the only templates we're using are the ones from DRF.

**Pull request merged**